File: frontend/src/contexts/backoffice/series/application/SerieSearcher.ts
import { Serie } from "@/contexts/backoffice/series/domain/Serie";
import { PaginatedResponse, SerieRepository } from "@/contexts/backoffice/series/domain/SerieRepository";

export class SerieSearcher {
    private readonly serieRepository: SerieRepository;

    constructor(serieRepository: SerieRepository) {
        this.serieRepository = serieRepository;
    }

    async run(): Promise<PaginatedResponse<Serie>> {
        return await this.serieRepository.matching();
    }
}

File: backend/src/contexts/backoffice/series/application/services/serie_searcher.py
from src.contexts.backoffice.series.domain import Serie, SerieRepository
from src.contexts.shared.domain.criteria import Criteria


class SerieSearcher:
    def __init__(self, repository: SerieRepository) -> None:
        self._repository = repository

    def run(self, filter: dict, sort: list, page_size: int, page_number: int) -> list[Serie]:
        criteria = Criteria.from_primitives(filter, sort, page_size, page_number)
        return self._repository.matching(criteria)
